tests/rptest/scale_tests/basic_auth_test.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import json
+import requests
+import threading
+from ducktape.mark import matrix
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.admin import Admin
+from rptest.services.redpanda import SecurityConfig
+from rptest.tests.pandaproxy_test import PandaProxyEndpoints, Consumer
+from rptest.util import wait_until_result
+from typing import Optional, List, Dict, Union
+
+
+class User:
+    def __init__(self):
+        self.username = f'user_{TopicSpec()._random_topic_suffix()}'
+        self.password = f'secret_{self.username}'
+        self.algorithm = 'SCRAM-SHA-256'
+
+
+class GetTopics(threading.Thread):
+    def __init__(self, thid: int, user: User, handle):
+        threading.Thread.__init__(self)
+        self.thid = thid
+        self.user = user
+        self._get_topics = handle
+        self.result_raw = None
+
+    def run(self):
+        self.result_raw = self._get_topics(auth=(self.user.username,
+                                                 self.user.password))
+
+
+class BasicAuthScaleTest(PandaProxyEndpoints):
+    topics = [
+        TopicSpec(partition_count=3),
+    ]
+
+    group = 'my_group'
+
+    def __init__(self, context):
+
+        security = SecurityConfig()
+        security.enable_sasl = True
+        security.endpoint_authn_method = 'sasl'
+        security.pp_authn_method = 'http_basic'
+
+        super(BasicAuthScaleTest, self).__init__(
+            context,
+            security=security,
+            pp_keep_alive=60000 * 5,  # Time in ms
+            pp_cache_max_size=1000)
```

Comment:
Currently dealing with a bug where the gate inside the client cache is closed when it should not be. The solution is to increase the cache max size to allow a greater number of unique clients in-memory. This is not a feasible solution though, the cache should be able to handle several thousand users without adjusting the max size.

The problem is likely due to the relationship between cache eviction and client stop. I need to investigate further.